compass/landice/tests/ismip6_forcing/atmosphere/create_mapfile_smb.py

```diff
@@ -0,0 +1,136 @@
+import os
+import subprocess
+import netCDF4
+import xarray as xr
+from mpas_tools.scrip.from_mpas import scrip_from_mpas
+
+
+# function that creates a mapping file from ismip6 grid to mali mesh
+def build_mapping_file(ismip6_grid_file, mapping_file, mali_mesh_file=None,
+                       method_remap=None):
+    """
+    Build a mapping file if it does not exist.
+
+    Parameters
+    ----------
+    ismip6_grid_file : str
+        ismip6 grid file
+    mapping_file : str
+        weights for interpolation from ismip6_grid_file to mali_mesh_file
+    mali_mesh_file : str, optional
+        The MALI mesh file if mapping file does not exist
+    method_remap : str, optional
+        Remapping method used in building a mapping file
+    """
+
+    if mali_mesh_file is None:
+        raise ValueError("Mapping file does not exist. To build one, Mali "
+                         "mesh file with '-f' should be provided. "
+                         "Type --help for info")
+
+    ismip6_scripfile = "temp_ismip6_8km_scrip.nc"
+    mali_scripfile = "temp_mali_scrip.nc"
+
+    # create the ismip6 scripfile if mapping file does not exist
+    # this is the projection of ismip6 data for Antarctica
+    print("Mapping file does not exist. Building one based on the "
+          "input/ouptut meshes")
+    print("Creating temporary scripfiles for ismip6 grid and mali mesh...")
+
+    # create a scripfile for the atmosphere forcing data
+    create_atm_scrip(ismip6_grid_file, ismip6_scripfile)
+
+    # create a MALI mesh scripfile if mapping file does not exist
+    scrip_from_mpas(mali_mesh_file, mali_scripfile)
+
+    # create a mapping file using ESMF weight gen
+    print(f"Creating a mapping file. Mapping method used: {method_remap}")
+
+    if method_remap is None:
+        raise ValueError("Desired remapping option should be provided with "
+                         "--method. Available options are 'bilinear',"
+                         "'neareststod', 'conserve'.")
+
+    args = ["ESMF_RegridWeightGen",
+            "-s", ismip6_scripfile,
+            "-d", mali_scripfile,
+            "-w", mapping_file,
+            "-m", method_remap,
+            "-i", "-64bit_offset",
+            "--dst_regional", "--src_regional"]
```

Comment:
On Anvil, I'm getting errors when running this because it needs to be called with `srun`:
```
Abort(1091087) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Init: Other MPI error, error stack:
MPIR_Init_thread(136): 
MPID_Init(950).......: 
MPIR_pmi_init(163)...: PMI_Init returned 14
Traceback (most recent call last):
  File "/home/ac.xylar/anvil/miniconda3/envs/compass_ismip6_forcing/bin/compass", line 33, in <module>
    sys.exit(load_entry_point('compass', 'console_scripts', 'compass')())
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/__main__.py", line 63, in main
    commands[args.command]()
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/run/serial.py", line 301, in main
    run_step(args.step_is_subprocess)
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/run/serial.py", line 262, in run_step
    test_case.run()
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/testcase.py", line 166, in run
    self._run_step(step, self.new_step_log_file)
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/testcase.py", line 282, in _run_step
    step.run()
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/landice/tests/ismip6_forcing/atmosphere/process_smb.py", line 106, in run
    self.remap_ismip6smb_to_mali(combined_file_temp, remapped_file_temp,
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/landice/tests/ismip6_forcing/atmosphere/process_smb.py", line 167, in remap_ismip6smb_to_mali
    build_mapping_file(input_file, mapping_file, mali_mesh_file,
  File "/gpfs/fs1/home/ac.xylar/mpas-work/compass/add_ismip6_forcing_testgroup/compass/landice/tests/ismip6_forcing/atmosphere/create_mapfile_smb.py", line 63, in build_mapping_file
    subprocess.check_call(args)
  File "/home/ac.xylar/anvil/miniconda3/envs/compass_ismip6_forcing/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ESMF_RegridWeightGen', '-s', 'temp_ismip6_8km_scrip.nc', '-d', 'temp_mali_scrip.nc', '-w', 'map_ismip6_8km_to_Antarctica_8to80km_bilinear.nc', '-m', 'bilinear', '-i', '-64bit_offset', '--dst_regional', '--src_regional']' returned non-zero exit status 1.
(compass_ismip6_forcing) [ac.xylar@b665 process_smb]$ which ESMF_RegridWeightGen
/lcrc/soft/climate/compass/anvil/spack/spack_for_mache_1.4.1/var/spack/environments/dev_compass_1_1_0-alpha_4_intel_impi/.spack-env/view/bin/ESMF_RegridWeightGen
```
Let's talk through how to do this.  It would also be an opportunity to set `cores` and `min_cores` and use a parallel call to `ESMF_RegridWeightGen`.

Reply:
The same will apply in `create_mapfile.py`.